apollo-client/src/main/java/com/ctrip/framework/apollo/internals/K8sConfigMapConfigRepository.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2022 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.internals;
+
+import com.ctrip.framework.apollo.kubernetes.KubernetesManager;
+import com.ctrip.framework.apollo.build.ApolloInjector;
+import com.ctrip.framework.apollo.core.ConfigConsts;
+import com.ctrip.framework.apollo.core.utils.DeferredLoggerFactory;
+import com.ctrip.framework.apollo.core.utils.StringUtils;
+import com.ctrip.framework.apollo.enums.ConfigSourceType;
+import com.ctrip.framework.apollo.exceptions.ApolloConfigException;
+import com.ctrip.framework.apollo.tracer.Tracer;
+import com.ctrip.framework.apollo.tracer.spi.Transaction;
+import com.ctrip.framework.apollo.util.ConfigUtil;
+import com.ctrip.framework.apollo.util.ExceptionUtil;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.slf4j.Logger;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * @author dyx1234
+ */
+public class K8sConfigMapConfigRepository extends AbstractConfigRepository
+        implements RepositoryChangeListener {
+    private static final Logger logger = DeferredLoggerFactory.getLogger(K8sConfigMapConfigRepository.class);
+    private final String namespace;
+    private String configMapName;
+    private String configMapKey;
+    private final String k8sNamespace;
+    private final ConfigUtil configUtil;
+    private final KubernetesManager kubernetesManager;
+    private volatile Properties configMapProperties;
+    private volatile ConfigRepository upstream;
+    private volatile ConfigSourceType sourceType = ConfigSourceType.CONFIGMAP;
+
+    /**
+     * Constructor
+     *
+     * @param namespace the namespace
+     */
+    public K8sConfigMapConfigRepository(String namespace) {
+        this(namespace, (ConfigRepository) null);
+    }
+
+    public K8sConfigMapConfigRepository(String namespace, KubernetesManager kubernetesManager) {
```

Comment:
It's not a practice to add a constructor only for test scenario. You could use `MockInjector.setInstance` method to inject the mocked KubernetesManager instance.